storagenode/gracefulexit/endpoint.go

```diff
@@ -89,40 +89,66 @@ func (e *Endpoint) GetNonExitingSatellites(ctx context.Context, req *pb.GetNonEx
 }
 
 // StartExit updates one or more satellites in the storagenode's database to be gracefully exiting.
-func (e *Endpoint) StartExit(ctx context.Context, req *pb.StartExitRequest) (*pb.StartExitResponse, error) {
-	e.log.Debug("initialize graceful exit: StartExit", zap.Int("satellite count", len(req.NodeIds)))
-	// save satellites info into db
-	resp := &pb.StartExitResponse{}
-	for _, satelliteID := range req.NodeIds {
-		e.log.Debug("initialize graceful exit: StartExit", zap.String("satellite ID", satelliteID.String()))
-
-		status := &pb.StartExitStatus{
-			Success: false,
-		}
-		domain, err := e.trust.GetAddress(ctx, satelliteID)
-		if err != nil {
-			e.log.Debug("initialize graceful exit: StartExit", zap.Error(err))
-			resp.Statuses = append(resp.Statuses, status)
-			continue
-		}
-		status.DomainName = domain
+func (e *Endpoint) StartExit(ctx context.Context, req *pb.StartExitRequest) (*pb.ExitProgress, error) {
+	e.log.Debug("initialize graceful exit: StartExit", zap.String("satellite ID", req.NodeId.String()))
 
-		// get space usage by satellites
-		spaceUsed, err := e.usageCache.SpaceUsedBySatellite(ctx, satelliteID)
-		if err != nil {
-			e.log.Debug("initialize graceful exit: StartExit", zap.Error(err))
-			resp.Statuses = append(resp.Statuses, status)
-			continue
+	domain, err := e.trust.GetAddress(ctx, req.NodeId)
+	if err != nil {
+		e.log.Debug("initialize graceful exit: StartExit", zap.Error(err))
+		return nil, errs.Wrap(err)
+	}
+
+	// get space usage by satellites
+	spaceUsed, err := e.usageCache.SpaceUsedBySatellite(ctx, req.NodeId)
+	if err != nil {
+		e.log.Debug("initialize graceful exit: StartExit", zap.Error(err))
+		return nil, errs.Wrap(err)
+	}
+
+	err = e.satellites.InitiateGracefulExit(ctx, req.NodeId, time.Now().UTC(), spaceUsed)
+	if err != nil {
+		e.log.Debug("initialize graceful exit: StartExit", zap.Error(err))
+		return nil, errs.Wrap(err)
+	}
+
+	return &pb.ExitProgress{
+		DomainName:      domain,
+		NodeId:          req.NodeId,
+		PercentComplete: float32(0),
+	}, nil
+}
+
+func (e *Endpoint) GetExitProgress(ctx context.Context, req *pb.GetExitProgressRequest) (*pb.GetExitProgressResponse, error) {
+	exitProgress, err := e.satellites.ListGracefulExits(ctx)
+	if err != nil {
+		return nil, errs.Wrap(err)
+	}
+
+	resp := &pb.GetExitProgressResponse{
+		Progress: make([]*pb.ExitProgress, 0, len(exitProgress)),
+	}
+	for _, progress := range exitProgress {
+		var percentCompleted float32
+		if progress.CompletionReceipt != nil {
+			percentCompleted = float32(100)
+		} else {
```

Comment:
i think we should remove this logic. it's possible that a node successfully exited (received completion receipt), but did not transfer all the pieces due to "acceptable" errors.  we should have another column that show's "successful?" as Y if the receipt exists.